Migrate card revisions to have join aliases in column_settings keys

calherries opened the pull request:
Epic: https://github.com/metabase/metabase/issues/30789
In depth description of the problem: https://github.com/metabase/metabase/pull/27487

Previously, I had created these two migrations that applied to report_card.visualization_settings: https://github.com/metabase/metabase/pull/30787 and https://github.com/metabase/metabase/pull/27487

But cards have revisions, so we need to migrate their revisions too. This PR creates two new migrations, each applying the same transformations in the above migrations to the revision table.

See the comments in the code for explanations about how these migrations work.

nemanjaglumac committed:
Remove CircleCI example from E2E dev guide (#31129)

calherries committed:
Handle case when uploads table is not found (#31065)

uladzimirdev committed:
Show name in save action form model picker (#30990)

Commit:
Add reverse migration for AddJoinAliasToVisualizationSettingsFieldRefs

kulyk committed:
[MLv2] Make stage index explicit (#31119)

* Make stage index required in MLv2 methods

* Add wrappers for `appendStage` and `dropStage`

* Fix notebook stage index management

* Fix `QuestionRowCount`

* Fix `StructuredQuery` limit method

* Clean redundant stages in notebook

bshepherdson committed:
[MLv2] Don't allow adding an expression with the same name as a column (#30809)

This was causing ambiguous references, since `order-by` clauses
reference columns from previous stages by name.

Fixes #30401.

tsmacdonald committed:
Use faster inserts in MySQL (#30820)

* Use faster inserts in MySQL

* Add local_infile check

* PR feedback

* Fix boolean handling for MySQL

* Fix tests; fix MySQL fallback if local_infile is turned off

* Turn local_infile on for MySQL tests

* Another way of doing MySQL local_infile :/

* sanitize-file-name was a misstep. Not needed for security and could cause problems

* Mistaken docstring

Commit:
CI: Avoid the deprecated set-output (#31146)

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Commit:
CI: Use updated GitHub actions to avoid deprecation warnings (#31148)

nllho committed:
docs - fix typo in db privileges (#31085)

nllho committed:
docs - update filters (#31101)

EmmadUsmani committed:
fix dashboard save failure after deleting tabs (#31039)

* fix dashboard save failure after deleting tabs

* fix undo regression and remove unused code in delete reducer case

metamben committed:
Add JS wrapper for selected-aggregation-operators (#31158)

Part of #31157.

escherize committed:
Add type column to collections + hydrate dashboards and cards with it (#30994)

* add migration for typed collections

* add hydration of collections to card+dashboard api

- adds it for:
  - GET /api/question/:id
  - GET /api/dashboard/:id

* fix test that checks for collection shape

* get tests working with hydrated collection

* typo fixes

* fix more tests

- PUT requests don't hydrate :collection

* update tests to fix param_values {} vs nil inconsistencies

bshepherdson committed:
[MLv2] Support `:case` expressions in aggregations (#31088)

Fixes #29935.

Part of the issue here was incorrect hand-rolling of the second
argument to `:case`, the default value. In the failing test case given
in #29935, it was written as `{:default 0}`. But that's accidentally
transferring for legacy MBQL works, where the default is an option and
the options come last rather than first in a clause.

This also fixes `lib.common/->op-arg` to make sure it recurses
properly. It wasn't recursing into lists or MBQL clauses, which left
unresolved `(fn [query stage] ...)` functions in eg. `:case`
alternatives, since those are nested inside lists.

kamilmielnik committed:
Fix - Cannot run native queries against MongoDB unless explicitly selecting the table (#31111)

* Update 15946 e2e test
- modal() expects ModalContent to be present in DOM, but NativeQuerymodal uses ModalBody directly
- add extra check for run button to be enabled

* Ensure collection is present in dataset query during query conversion

* Unskip e2e test for #15946

* Handle string tableId

ranquild committed:
Add entity loader hooks for Search, RecentItems, PopularItems, DatabaseCandidates (#31143)

Commit:
Limit selections to query_type="query"

Commit:
Make like strings more specific

Commit:
Add migration

Commit:
Add migration change set

Commit:
Fix where clause

Commit:
Further optimize where clause

Commit:
Add migration for legacy column field refs

Commit:
Rename and refactor

Commit:
Reuse remove-join-alias-from-column-settings-field-refs

Commit:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs

Commit:
Fix for MySQL

Commit:
Fix for MySQL

Commit:
Restore log/info

Commit:
Fix migration

Commit:
Add where clause for perf optimization

Commit:
Fix MigrateLegacyColumnSettingsFieldRefs and add where clause optimization to MigrateLegacyResultMetadataFieldRefs

Commit:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs

calherries commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -352,3 +361,88 @@
+                                                                         [(json/generate-string ["ref" ["field" id (assoc opts "join-alias" alias)]]) v])
+                                                                       _ '()))
+                                                                   column_settings))]
+                          ;; existing column settings should take precedence over the copies in case there is a conflict
```
Small detail: the case of a conflict should only be possible if the revision happened since the `AddJoinAliasToVisualizationSettingsFieldRefs` migration (v47.00-028), but we handle it anyway just in case

calherries commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -352,3 +361,88 @@
+                                                     [:like :object "%ref\\\\\\\",[\\\\\\\"field-literal%"]
+                                                     [:like :object "%ref\\\\\\\",[\\\\\\\"fk->%"]]]}))))
+
+(define-reversible-migration RevisionAddJoinAliasToColumnSettingsFieldRefs
```
Explanation of how this migration works:

For each alias in `(get-in card ["dataset_query" "query" "joins"])`, we add new entries in `column_settings`.

For example:

```
If this was a entry in column_settings before the migration:
["ref" ["field" id nil]] {:column_title "A"}

And the query had 2 aliases in the query:
"Joined table 1", "Joined table 2"

After migration, there would be 3 entries in the column_settings table for each entry before the migration:
["ref" ["field" id nil]] {:column_title "A"}
["ref" ["field" id {"join-alias" "Joined table 1"}]] {:column_title "A"}
["ref" ["field" id {"join-alias" "Joined table 2"}]] {:column_title "A"}
```

Even if `["field" id {"join-alias" "Joined table 1"}]]` and `["field" id {"join-alias" "Joined table 2"}]]` aren't actually field refs in the card, this method preserves the observed column settings from the perspective of the user. At least one of them will be a real field ref of the card. The ones that aren't don't matter.

qnkhuat commented on resources/migrations/000_migrations.yaml:
```diff
@@ -14812,6 +14812,23 @@ databaseChangeLog:
 
+  - changeSet:
+      id: v47.00-033
+      author: calherres
```
```suggestion
      author: calherries
```

qnkhuat commented on resources/migrations/000_migrations.yaml:
```diff
@@ -14812,6 +14812,23 @@ databaseChangeLog:
+
+  - changeSet:
+      id: v47.00-034
+      author: calherres
```
```suggestion
      author: calherries
```

qnkhuat commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -188,29 +188,37 @@
-        (json/generate-string))))
+    (m/update-existing viz-settings "column_settings" update-keys
+                       (fn [k]
+                         (-> k
+                             (json/parse-string)
+                             (vec)
+                             (old-to-new)
+                             (json/generate-string))))))
```
```suggestion
                         (-> k
                             json/parse-string
                             vec
                             old-to-new
                             json/generate-string)))))
```

Commit:
Undo renaming migration

Commit:
Fix typos

Commit:
Style fixes

qnkhuat commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -352,3 +361,88 @@
+                                                 [:like :object "%joins%"]
+                                                 [:= :model "Card"]]})))
+  ;; Reverse migration
+  (let [update-one!
```
I might miss the context behind this from the previous PRs, but why do we need to reverse it?

calherries replied:
Good q. You could change my mind on whether we should include this or not. We actually don't _have_ to reverse it, since there will always be one entry in column_settings with no join-alias that is identical to the before the migration. But I thought we should avoid having extra entries if we can. It's unlikely to cause any issues either way though.

qnkhuat reviewed (approved):
as far as I can tell this makes sense to me, have a note about adding a checksum for an old migration we edited in this PR

but would be much better if Cam can have a look at this.

calherries commented on src/metabase/db/custom_migrations.clj:
```diff
@@ -352,3 +361,88 @@
+        update-one!
+        (fn [revision]
+          (let [card (json/parse-string (:object revision))]
+            (when (not= (get card "query_type") "native") ; native queries won't have join aliases, so we can exclude them straight away
```
Unfortunately query_type is nullable, so these aren't equivalent. If query_type=nil, we do not know whether the dataset_query is native or MBQL. [some context on slack](https://metaboat.slack.com/archives/CKZEMT1MJ/p1685546996307979)

npfitz reviewed (approved):
I did a test where I:
- rolled back locally and started with a fresh DB, created a question with some revisions, 
- rolled up to master and saw that the question had migrated, but none of the revisions had. applied an old revision and changes were only applied to half the columns
- rolled up to this commit and saw revisions had join aliases in the column settings now 🙌 

With that said, I think this is working correctly!

Commit:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs

calherries commented on resources/migrations/000_migrations.yaml:
```diff
@@ -14776,19 +14776,19 @@ databaseChangeLog:
       changes:
         - customChange:
-            class: "metabase.db.custom_migrations.AddJoinAliasToColumnSettingsFieldRefs"
+            class: "metabase.db.custom_migrations.AddJoinAliasToVisualizationSettingsFieldRefs"
```
This branch matched master here already, I just merged with master and the diff disappeared

Commit:
Tidy

Commit:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs

calherries commented:
@qnkhuat I checked in with Cam https://metaboat.slack.com/archives/C04DN5VRQM6/p1686674127331389?thread_ts=1686660217.763829&cid=C04DN5VRQM6, so I'll merge this

calherries committed:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs

nemanjaglumac committed:
Refactor E2E tests in the `sharing` folder (#31455)

* Extract `questionDetails`

* Move `public-sharing` to the `sharing` folder

* Remove redundant test

This part was covered in `public-sharing.cy.spec.js`.

* Create a separate `public-dashboard` E2E spec

- Extract relevant part from `public.cy.spec.js`
- Make all tests in `public-dashboard.cy.spec.js` run in isolation

* Make tests in `public.cy.spec.js` run in isolation

* Remove redundant wait

* Limit the query results to speed test up

* Merge public questions E2E tests

* Merge `downloads` with the `sharing` E2E group

* Remove `downloads` from the folder list

* Refactor `public-dashboard.cy.spec.js`

- Clean up linter warnings by using semantic selectors
- Speed test up by using API

escherize committed:
Auditv2 modify views: add v_audit_log + update v_content (#31563)

* add the migrations

* edit comment

* add 'if exists' to drop view statements

* update yaml: move sql to the right spot

nemanjaglumac committed:
[E2E] Do not hardcode admin id (#31576)

We're trying to minimize the amount of hard coded IDs (or any other data) in E2E tests. This change doesn't assume admin's ID is always going to be 1. It fetches the id from the current user instead.

This should make test more resilient to change.

shaunlebron committed:
fix frontend dev for safari (#31457)

shaunlebron committed:
fix funnel tooltip artifacts in safari (#31494)

* fix funnel tooltip artifacts in safari

* force composite-layer instead of forceRedraw

* used styled components

---------

Co-authored-by: Aleksandr Lesnenko <alxnddr@gmail.com>

kulyk committed:
Migrate notebook breakout step to MLv2 (#30693)

* Add `BreakoutClause` type

* Extend `displayInfo` for breakouts

* Add wrappers for `breakouts` and `breakout`

* Fix property names in breakout clause types

* Add TypeScript wrappers for temporal buckets

* Add basic `ColumnPrecisionPickerPopover`

* Add TypeScript wrappers for binning strategies

* Add `BinningStrategyPickerPopover`

* Rename to `TemporalBucketPickerPopover`

* Remove type properties from breakout clause type

* Unify temporal bucket and binning strategy types

* Unify bucket pickers

* Fix `isCalculated` property name

* Fix bucket methods usage in Jest

* Allow calling `displayInfo` with `Clause`

* Add `isClauseColumn` helper

* Make bucket pickers optional in `QueryColumnPicker`

* Highlight selected column in `QueryColumnPicker`

* Add a11y labels to bucket pickers

* Use `createMockMetadata` in notebook tests

* Use MLv2 in `BreakoutStep`

* Update `SortStep`

* Fix stage index for `BreakoutStep`

* Tweak bucket types

* Use named exports for bucket popovers

* Format temporal bucket name in the column picker

* Update `binning` and `temporalBucket` types

* Add `isSameBucket` utility

* Tweak the column picker

* highlight selected bucket
* prevent unexpected bucket reset

* Use temporal buckets JS wrappers

* Use default bucket in the column picker

* Fix temporal bucket test

* Add test IDs for compatibility with e2e tests

* Fix "Unbinned" bucket handling

* Fix e2e test to use a11y selectors

* Add long-display-name to display-info

* Use long display name in the notebook

* Skip getting metadata from nil

* Handle JS objects as fingerprint metadata

* Set max height for bucket picker popover

* Fix `summarization.cy.spec` test

* Fix `joins.cy.spec` test

* Attempt to fix MLv2 query validation

* Test that integer expressions are recognized by isa/numeric?

* Make stage index explicit

* Fix custom-columns e2e tests

* Fix breakout validation

* Remove `React` imports

* Break down `hasBucketing` prop

* Keep the metadata from the tables list for questions (#31162)

* Keep the metadata from the tables list for questions

Previously, the metadata loaded from "questions" would overwrite
the metadata loaded from "tables" if both existed. This PR changes
that to merge the information from the two sources instead.

* Skip using _plainObject when parsing cards from table metadata

* Format saved question display names in joins as "Question <card-id>"

* Deal with field-ref shaped field IDs in the JS provided metadata

* Fix long display name source question breakout (#31312)

* Fix icon import

* Fix e2e test

* Fix e2e test

* Allow field references by name

* Make sure no temporal buckets and binning strategies are offered for expressions

* Fix MLv2 tests

* Return empty vector for visible-columns on nil

* Add `base-type` to MLv1 compatible field rfs

* Use `selected` flag on buckets

* Recognize expression columns when marking selected breakouts

* Add backend regression test for expression breakout column handling

* Test fix

* Rework column bucket pickers

* delegate more responsibilities to binning/temporal pickers
* include "Don't bin" option on FE vs BE

* Fix e2e test

* Rework bucket pickers 2

Forgot to push this in the first place 🤦‍♂️

* Fix redundant eslint-disable comment

* Use *-position to check if a column is selected

* Remove dont-bin pseudo binning strategy (#31443)

* Fix matching breakout columns with binning (#31441)

* Stop filtering out "Don't bin" on the FE

* Fix #31482 not showing join display name for busted breakout from drill thru

* MLv2: Fix matching of field literals to field integer IDs (#31487)

* MLv2: Fix matching of field literals to field integer IDs

* Better test

* Update src/metabase/lib/equality.cljc

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

---------

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

* Remove infinite recursion when resolving fields by name (#31485)

* Remove infinite recursion when resolving fields by name

* Remove stray tap>

---------

Co-authored-by: Case Nelson <case@metabase.com>
Co-authored-by: Tamás Benkő <tamas@metabase.com>
Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>
Co-authored-by: Cam Saul <github@camsaul.com>
Co-authored-by: Cam Saul <1455846+camsaul@users.noreply.github.com>

shaunlebron committed:
Fix boolean icons in SQL query editor (#31538)

* Fetch field icon from dimension instead of semantic_type (#28622)

* simplify field.dimension().icon() → field.icon()

* resync tooltip to reflect non-semantic icon

* add ts-ignore to iconName with justification

* Updating eslint rules to allow importing types to metabase-lib

* Revert "Updating eslint rules to allow importing types to metabase-lib"

This reverts commit 1b6aca4362a7ca530046466bc9c1bfc02d5b116c.

---------

Co-authored-by: Abdul Aziz Ali <abdulazizali@acm.org>
Co-authored-by: Nick Fitzpatrick <nickfitz.582@gmail.com>

deniskaber committed:
Fix dashboard description markdown overflow (#31534)

* Fix dashboard description markdown overflow

* Review fixes, Improved e2e test

escherize committed:
update audit log view migration (#31589)

* update audit log view migration

* separate mysql and fix h2

* typo

* fix views

---------

Co-authored-by: Luiz Arakaki <luiz.arakaki@metabase.com>

darksciencebase committed:
Revamp and split an internal issue template (#31575)

Previously: one “feature or project” template with the `.Epic` label automatically attached.
Now: two templates, one for epics, one for smaller things that don't require milestones (or `.Epic` label; they get `.Task` instead).

Squashed commit messages follow:

* Remove unused optional field

* Update title to current convention

* Fix typo

* Remove unused section

* Change capitalization for consistency

* Remove redundant description

* Rename the issue template to mention epic

* Shorten the description of the issue template

* Lighten up the milestones section

Unfortunately, it doesn't seem possible to prepopulate the issue with an
empty tasklist (https://github.com/orgs/community/discussions/50502).

* Add an issue template for tasks [ci skip]

We had one for epics, with milestones and `.Epic` label. Pretty often
that label was accidentally left on smaller issues that were created
using the same template. Now the smaller issues have their own template
and their own label. Yay.

https://metaboat.slack.com/archives/C04CGFHCJDD/p1686548875239219

* Remove redundant angle brackets

* Use a string to get a valid yaml file

Without the quotes, github says
`Error in user YAML: (<unknown>): did not find expected key while parsing
a block mapping at line 1 column 1`

* Remove rarely if ever used placeholders

* Call out optional-ness of some links

* Contextualize a header

Context can be some text, a slack thread, a notion page, another issue,
whatever.

tsmacdonald committed:
[CSV] Avoid expensive call to fetch DB metadata (#31556)

* Avoid expensive call to fetch DB metadata

Double-checking the schema name is unnecessarily defensive since we
already call include-schema?

* Check that schema is set

calherries committed:
Add join alias to dashboard card visualization_settings.column_settings field refs (#31476)

john-metabase committed:
Filter import on YAML extension (#31477)

* Filter import on YAML extension

* sorts namespaces

---------

Co-authored-by: Luiz Arakaki <luiz.arakaki@metabase.com>

metamben committed:
Fixes #31555. (#31595)

Also make sure that the attached columns have the right temporal bucket information.

devurandom committed:
Respect MB_COLORIZE_LOGS and MB_EMOJI_IN_LOGS environment variables (#31612)

Due to AOT (Ahead Of Time) compilation (specifically direct linking),
the `MB_COLORIZE_LOGS` and `MB_EMOJI_IN_LOGS` env vars where not
respected.

Marking `metabase.util.format/colorize` and `metabase.util/emoji` with
`^:redef` metadata prevents the compiler from using direct linking for
calls to these functions (and thus evaluating the env vars at compile
time) and thus delays resolution of what function to call until runtime
when the actual user-provided value of the env var is known.
cf. https://clojure.org/reference/compilation#directlinking

Closes: https://github.com/metabase/metabase/issues/30518

metamben committed:
Allow generating metadata for tables with mil columns (#31618)

This should not really happen, but parts of the code handle this
situation while others don't.

john-metabase committed:
Fix Snowflake sync with uploaded private key (#31593)

metamben committed:
Disable slow test to prevent flakiness (#31594)

* Disable slow test to prevent flakiness

Fixes #29266.

qnkhuat committed:
Rest of models to toucan2 🎉 (#31385)

* User to toucan2

* Pulse to toucan2

* PulseCard to toucan 2

* PulseChannel and PulseChannelRecipient to toucan2

* Query, QueryCache, QueryExecution to toucan2

* Metric, Segment, MetricImportantField to toucan2

* ParameterCard to toucan2

* DataMigrations and FakeCard to toucan2

* NativeQuerySnippet to toucan2

qnkhuat committed:
Do not record public info in revisions (#31598)

npfitz committed:
custom homepage modal (#31398)

* WIP showing Toast

* add dismissed_toasts stub on GET api/user/current

* backend to save dismissed toast names as user-local-settings

- adds dismissed_toasts key to GET user/current
- introduces POST /user/:user-id/dismissed/:toast-name
  - For now, only custom_homepage_changed is a valid `toast-name`, but
    it can be extended to others.
  - I nearly put in a DELETE to unset it, which could be useful for
    tests, but took it out.

* add untested DELETE route to ease devtime pains

* setup dismissed-custom-dashboard-toast to be "yet another setting"

- remove specific endpoints
- keep the result on GET api/user/current

* dismissed_toasts + improve setting api to take Snake Case

- with tests

* Adding dismiss action

* dismissed toast map

* Ensure that toast is dismissed for admin who set the custom homepage property

* Fixing 2 toasts on login

* Moving dismissed toast to settings

* updateSettings, undo.canDismiss

* adding loading state to settings, removing unneeded CLJ tests


---------

Co-authored-by: Bryan Maass <bryan.maass@gmail.com>

nemanjaglumac committed:
[E2E] Add `actions` to stubbed skipped job (#31640)

I forgot to do this in #31432.
As the result, some PRs have been stuck in CI due to the required check.

This PR will run the fallback job for `actions` E2E group when changed
files don't require E2E tests to run.

[ci skip]

nemanjaglumac committed:
Fix wrong issue reference in E2E test (#31547)

[ci skip]

john-metabase committed:
Always create a transient dataset for BigQuery tests (#31605)

Resolves #28792 

* Always create a transient dataset for BigQuery tests

* Simplifies BigQuery test data code

* More BigQuery test data code cleanup

* More BigQuery test data code cleanup, test fix

* Extracts test-db-name in bigquery tests, restores normalize-name

* Fixes BigQuery QP tests for new test DB name scheme

* Removes extraneous clojure.core

* Updates bigquery QP test helper code style

* Cleans up comments and code in bigquery test data

* Update modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

---------

Co-authored-by: metamben <103100869+metamben@users.noreply.github.com>

jeff-bruemmer committed:
[Docs] Update `running-metabase-on-docker.md` (#31516)

jeff-bruemmer committed:
[Docs] Update data reference (#31526)

* update data reference

* update screenshot

metamben committed:
Add test after speculative fix (#31633)

Commit:
Update default and min size constraints for dashboard cards (#31546)

nemanjaglumac committed:
[CI] Solve OOM failures for `be-cloverage` job (#31644)

* Use BuildJet runner for `be-cloverage`

We're seeing OOM failures in `be-linter-cloverage` job in CI.

* revert back to using ubuntu-22.04 and increase heap size to 2GB

* 2gb to 4gb

* add comment

---------

Co-authored-by: Ngoc Khuat <qn.khuat@gmail.com>

jeff-bruemmer committed:
docs - csv uploads (#31629)

qwef committed:
Fix some typos in the api/user logic (#31667)

metamben committed:
Ensure an alias is set when creating a join (#31568)

* Calculate and set join alias when joining

camsaul committed:
MLv2 implement `suggested-join-condition` (#31590)

* Implement `suggested-join-condition`

* Update docstring

iethree committed:
[E2E] Test Entity ID Agnosticism (#31579)

* load snapshot instance data

* stop hardcoding question/1

* parameterize default question IDs

* cleanup

* fix imports

* more slashes

* misc fixes

---------

Co-authored-by: Bryan Maass <bryan.maass@gmail.com>

qnkhuat committed:
Permissions and related tables to toucan2 (#30726)

* Permissions to toucan2

* PermissionsGroup to toucan2

* PermissionsGroupMembership to toucan2

* PermissionsRevision to toucan2

* GroupTableAccessPolicy to toucan2

* ApplicationPermissionsRevision to toucan2

* CollectionPermissionGraphRevision to toucan2 too

qnkhuat committed:
Collection, CollectionPermGraphRevision to toucan2 (#30724)

* Collection to toucan2

camsaul committed:
Replace `connection-with-timezone` with `do-with-connection-with-timezone` (#22166)

* Squash

* Replace connection-with-timezone usages

* Add deprecation warning.

* Fix docstring

WiNloSt committed:
Merge Encourage admins to add a database to master (#31610)

* Encourage admin users to connect to a database (#31210)

* Encourage admin users to connect to a database

* Only render either payment banner or database prompt banner

Discussed here:
https://metaboat.slack.com/archives/C057T1QTB3L/p1685549180307149?thread_ts=1685547913.580079&cid=C057T1QTB3L

* Improve mobile view layout

* Address review: Remove `withRouter` HoC

* Address review: Remove optional chaining

* Address review: Refactor a component's deprecated `shouldRender` property

* Add an e2e test to cover happy path

* Remove new translation in `aria-label`

* Fix sticky parameter widgets when showing database prompt banner

* Fix E2E tests

* Address review: Add missing unit tests

* Fix a failed E2E test from rebasing off `master` branch

* Add analytics (#31470)

* 2/x Add analytics

* Fix a failed E2E test from merging code from master

* Prevent rendering the database prompt banner when white labeling (#31471)

* 3/x Convert enterpise settings selectors to TypeScript

* Add missing enterprise settings selectors tests

* Don't render db prompt banner when white labeling

* Reorganize the test to make an EE test more standing out

* Address review: Remove unused function

* 4/x Follow up reviews and tests (#31545)

* 4/x Address review: Simplify code structure

* Address review: Fix variable name collision

* Move logic to display database banner to a hook

* Remove unnecessary await

* Restructure AppBanner tests

* Swap test position to match the defined test matrix

* Address review: removed outdated comment

* Address review: Make `useShouldShowDatabasePromptBanner` hook tests clearer

* Address review: Make conditions easier to follow

Co-authored-by: Uladzimir Havenchyk <uladzimir.dev@gmail.com>
Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>

---------

Co-authored-by: Uladzimir Havenchyk <uladzimir.dev@gmail.com>
Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>

* 5/x Add embedding tests (#31557)

* 5/x Add embedding tests

* Clean up: Fix a selector signature

* Address review: Using existing utility

* Ensures that embedding pages are loaded before asserting UI not existing

* Ready to write PublicQuestion tests

* Move public and signed embeds tests to unit tests

* Removed outdated eslint comment

---------

Co-authored-by: Uladzimir Havenchyk <uladzimir.dev@gmail.com>
Co-authored-by: Kamil Mielnik <kamil@kamilmielnik.com>

qnkhuat committed:
Adding new db trigger full scan by default (#31572)

* Adding new db trigger full scan by default

* Add DatabaseForm unit test
---------

Co-authored-by: Mahatthana Nomsawadi <mahatthana.n@gmail.com>

Commit:
Merge branch 'master' into revision-add-join-alias-to-visualization-settings-field-refs
